.github/workflows/ci.yml

```
@@ -13,7 +13,35 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
   cancel-in-progress: true
 jobs:
+  deps:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+      - name: Install platform dependencies
+        run: |
+          sudo apt -y update
+          sudo apt -y install libcurl4-openssl-dev libssl-dev pkg-config libxml2-dev libxslt-dev
+      - uses: actions/setup-python@v4
+        with:
+          python-version-file: '.python-version'
+      - name: pip cache
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: pip-${{ runner.os }}
+      - name: Install Python dependencies
+        run: |
+          pip install -U setuptools wheel
+          pip install --no-cache-dir  'pip @ git+https://github.com/sbidoul/pip@49d3354e1c807bafdd715225d2aaa5cfc3532434'
```

```suggestion
          pip install  'pip @ git+https://github.com/pypa/pip@refs/pull/11897/head'
```

This should not change anything but at least you'll get the latest version of the PR.